pkg/machine/ignition/ignition.go

```diff
@@ -685,6 +685,50 @@ done
 `
 }
 
+func (i *IgnitionBuilder) AddPlaybook(contents string, destPath string, username string) error {
+	// create the ignition file object
+	f := File{
+		Node: Node{
+			Group: GetNodeGrp(username),
+			Path:  destPath,
+			User:  GetNodeUsr(username),
+		},
+		FileEmbedded1: FileEmbedded1{
+			Append: nil,
+			Contents: Resource{
+				Source: EncodeDataURLPtr(contents),
+			},
+			Mode: IntToPtr(0744),
+		},
+	}
+
+	// call ignitionBuilder.WithFile
+	// add the config file to the ignition object
+	i.WithFile(f)
+
+	unit := parser.NewUnitFile()
+	unit.Add("Unit", "After", "ready.service")
+	unit.Add("Service", "Type", "oneshot")
+	unit.Add("Service", "User", username)
+	unit.Add("Service", "Group", username)
+	unit.Add("Service", "ExecStart", fmt.Sprintf("ansible-playbook %s", destPath))
+	unit.Add("Install", "WantedBy", "default.target")
```

Comment:
> and execute it after the first boot.

So the docs the say this only runs once on first boot. But this is a normal enabled unit which will be triggered on every boot.
Unless I am missing something goes for WSL where you ssh on each start?

I don't think we want to run the playbook on each start, that seems wrong.

Reply:
You're right. I'll add a check to see if it's the first boot and only add the unit file/pipe then

Reply:
I think (untested)
`ConditionFirstBoot=yes` https://www.freedesktop.org/software/systemd/man/latest/systemd-firstboot.html

For WSL we should have some state in the config to know if this is first boot somewhere

Reply:
yep, just saw the same thing systemd item. thanks!

Reply:
Should be fixed now.